apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/util/DatabaseTestUtils.java

```diff
@@ -0,0 +1,427 @@
+package gov.cms.bfd.migrator.util;
+
+import com.google.common.collect.ImmutableList;
+import gov.cms.bfd.sharedutils.exceptions.BadCodeMonkeyException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.net.Inet4Address;
+import java.net.ServerSocket;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Properties;
+import javax.sql.DataSource;
+import org.flywaydb.core.Flyway;
+import org.hsqldb.jdbc.JDBCDataSource;
+import org.hsqldb.persist.HsqlProperties;
+import org.hsqldb.server.ServerAcl.AclFormatException;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides utilities for managing the database in integration tests.
+ *
+ * <p>Note: This is placed in <code>src/main/java</code> (rather than <code>src/test/java</code>)
+ * for convenience: test dependencies aren't transitive, which tends to eff things up.
+ */
+public final class DatabaseTestUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseTestUtils.class);
+
+  /**
+   * This fake JDBC URL prefix is used for custom database setups only used in integration tests,
+   * e.g. {@link #initUnpooledDataSourceForHsqlEmbeddedWithServer(String)}.
+   */
+  public static final String JDBC_URL_PREFIX_BLUEBUTTON_TEST = "jdbc:bfd-test:";
+
+  /**
+   * We need to inform Flyway of all of our schemas in order for {@link Flyway#clean()} to work
+   * properly.
+   *
+   * <p>PUBLIC for hsql and public for postgres since they differ in caps sensitivity.
+   */
+  public static final ImmutableList<String> FLYWAY_CLEAN_SCHEMAS =
+      ImmutableList.of("public", "PUBLIC", "pre_adj");
```

Comment:
not sure of the mixed case schema names (i.e., one uses all caps, one all lower_case). The database is agnostic to case unless explicitly specified as such (i.e., `Beneficiaries`). so for the schema names they could have been all upcase, all locase, or even some crazy mixed case, and they all mean the same.

Reply:
The test was not clearing the DB correctly for postgres specifically (when using` flyway.clean()`) if the DB name here was not lowercase. This caused me hours of pain.